components/teams-service/storage/postgres/postgres.go

```diff
@@ -362,24 +365,62 @@ func (p *postgres) PurgeUserMembership(ctx context.Context, userID string) ([]st
 // from a different context we should apply project filtering in this function as well.
 // Not doing it to save us a database call since it's not needed currently.
 func (p *postgres) AddUsers(ctx context.Context,
-	id string, userIDs []string) (storage.Team, error) {
-	var t storage.Team
-	err := p.db.QueryRowContext(ctx,
-		`WITH moved_row_ids AS (
-			INSERT INTO teams_users_associations (team_db_id, user_id, created_at)
+	id string, userIDs []string) (updatedUserIDs []string, err error) {
+
+	tx, err := p.db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, p.processError(err)
+	}
+
+	projectsFilter, err := ProjectsListFromContext(ctx)
+	if err != nil {
+		return nil, p.processError(err)
+	}
+
+	var dbID int
+	var teamProjects []string
+	// ensure the team exists and isn't filtered out by the project filter
+	err = p.db.QueryRowContext(ctx,
+		`SELECT db_id, projects FROM teams
+		WHERE id = $1 AND projects_match(projects, $2::TEXT[])`,
+		id, pq.Array(projectsFilter)).
+		Scan(&dbID, pq.Array(&teamProjects)) // TODO is there a way to get these projects without returning them?
```

Comment:
oh yeah I had a question if there was another way to write this query that gets the team's projects for the projects_match but doesn't return the projects (since we just need them for that check)

Reply:
Clarity might be subjective, I find (1.) clearer. Even more so if we had a `team_projects(id)` helper, maybe:
```
SELECT db_id FROM teams WHERE id = $1 AND
  projects_match(team_projects(id), $2::TEXT[]);
```
But I can't shake the feeling that this could be simpler.

Doesn't this do the trick?
```
SELECT db_id FROM teams
WHERE id = $1 AND projects_match(projects, $2::TEXT[])
```
-- it's your query without returning the `projects` (from the `SELECT`) -- after all, `projects` is a column on `teams`, so this should be valid.

Reply:
oh I thought I'd tried this and the tests failed 🤔 but it worked so must have been mislead by something else! 
Thanks! :)